source/token_provider_sso_profile.c

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/auth/credentials.h>
+
+#include <aws/auth/private/aws_profile.h>
+#include <aws/auth/private/credentials_utils.h>
+#include <aws/auth/private/sso_token_providers.h>
+#include <aws/auth/private/sso_token_utils.h>
+#include <aws/common/clock.h>
+
+#ifdef _MSC_VER
+/* allow non-constant declared initializers. */
+#    pragma warning(disable : 4204)
+#endif
+
+/*
+ * sso-token profile provider implementation
+ */
+struct aws_token_provider_profile_impl {
+    struct aws_string *token_file_path;
+
+    aws_io_clock_fn *system_clock_fn;
+};
+
+static int s_token_provider_profile_get_token_async(
```

Comment:
why naming it `*_async`, there is no async call in here?

Reply:
Thanks, fixed.